### PR TITLE
Complex refactoring

### PR DESCRIPTION
**Pull request by roxblnfk**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️

`CycleORMHelper` removed
Added `cycle/schema` command 
Added SQL logger support; added temporary StdoutLogger
`migrate/up` and `migrate/down` now dispatch events

### CODE REVIEW

File: src/Command/SchemaCommand.php [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Yiisoft\Yii\Console\ExitCode;
+
+class SchemaCommand extends BaseMigrationCommand
+{
+    protected static $defaultName = 'cycle/schema';
+
+    private SchemaInterface $schema;
+    private array $strRelations = [
```
Comment (rustamwin) [inline src/Command/SchemaCommand.php]: Should be static?

File: src/Factory/SchemaFromConveyorFactory.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+     * @param ContainerInterface $container
+     * @return Schema
+     * @throws \Psr\SimpleCache\InvalidArgumentException
+     * @throws \Yiisoft\Yii\Cycle\Exception\BadGeneratorDeclarationException
+     */
+    public function __invoke(ContainerInterface $container)
+    {
+        $schemaArray = null;
+        if ($this->cacheEnabled) {
+            $container->get(CacheInterface::class)->get($this->cacheKey);
```
Comment (thenotsoft) [inline src/Factory/SchemaFromConveyorFactory.php]: probably should be like that?

`$schemaArray = $container->get(CacheInterface::class)->get($this->cacheKey);`

File: src/Command/SchemaCommand.php (hunk 1 above)
Comment (roxblnfk) [reply src/Command/SchemaCommand.php]: will be const

**Comment by roxblnfk** [comment]
What you phink about `cycle/schema` command?
Output now looks like this:
image

File: src/Command/SchemaCommand.php (hunk 1 above)
Comment (rustamwin) [reply src/Command/SchemaCommand.php]: Yeah, that's what I meant :smile:

**Comment by rustamwin** [comment]
> What you phink about cycle/schema command?

LGTM. Can you change the color red? It looks a little awful

**Comment by roxblnfk** [comment]
> > What you phink about cycle/schema command?
> 
> LGTM. Can you change the color red? It looks a little awful

magenta?

image
```
classname/value type -> blue
batabase (BD name, table, field)-> green
role -> magenta
entity property -> cyan
```

**Comment by thenotsoft** [comment]
> > > What you phink about cycle/schema command?
> > 
> > 
> > LGTM. Can you change the color red? It looks a little awful
> 
> magenta?
> 
> image
> 
> ```
> classname/value type -> blue
> batabase (BD name, table, field)-> green
> role -> magenta
> entity property -> cyan
> ```

like magento look better

**Comment by roxblnfk** [comment]
Magenta ok
---
I also added SQL logger support and temporary `StdoutLogger`\
For each request `StdoutLogger` will print SQL queries in the console if you started your application with `yii serve`\
In the future `StdoutLogger` will be removed (when we will have debug-tools). You can also pass other `LoggerInterface` definition

**Comment by thenotsoft** [comment]
@roxblnfk What do you think about this? https://github.com/cycle/orm/issues/59

**Comment by roxblnfk** [comment]
> @roxblnfk What do you think about this? [cycle/orm#59](https://github.com/cycle/orm/issues/59)

We discussed this issue in the discord [channel](https://discord.gg/AVvTEN). Probably @wolfy-j will add a cache dependense to the PG driver

**Comment by wolfy-j** [comment]
Would PSR-16 work for these purposes?

**Comment by samdark** [comment]
> Would PSR-16 work for these purposes?

Yes.

File: src/Command/BaseMigrationCommand.php [hunk 3]
```diff
@@ -11,14 +11,12 @@
...
-use Yiisoft\Yii\Cycle\Helper\CycleOrmHelper;
 
 abstract class BaseMigrationCommand extends Command
 {
     protected DatabaseManager $dbal;
     protected MigrationConfig $config;
     protected Migrator $migrator;
-    protected CycleOrmHelper $cycleOrmHelper;
 
     protected static $migrationStatus = [
```
Comment (samdark) [inline src/Command/BaseMigrationCommand.php]: Is it meant to be extended? Could it be constant? Also `array` could be specified as type.

File: src/Command/DownCommand.php [hunk 4]
```diff
@@ -2,15 +2,34 @@
...
 use Spiral\Migrations\MigrationInterface;
+use Spiral\Migrations\Migrator;
+use Spiral\Migrations\State;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 use Yiisoft\Yii\Console\ExitCode;
+use Yiisoft\Yii\Cycle\Event\AfterMigrate;
+use Yiisoft\Yii\Cycle\Event\BeforeMigrate;
 
 class DownCommand extends BaseMigrationCommand
```
Comment (samdark) [inline src/Command/DownCommand.php]: `final`?

File: src/Command/GenerateCommand.php [hunk 5]
```diff
@@ -2,18 +2,36 @@
...
 use Symfony\Component\Console\Input\StreamableInputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\Question\ConfirmationQuestion;
 use Symfony\Component\Console\Question\Question;
 use Yiisoft\Yii\Console\ExitCode;
-use Yiisoft\Yii\Cycle\Generator\ShowChangesGenerator;
+use Yiisoft\Yii\Cycle\Conveyor\SchemaConveyorInterface;
+use Yiisoft\Yii\Cycle\Generator\PrintChanges;
 
 class GenerateCommand extends BaseMigrationCommand
```
Comment (samdark) [inline src/Command/GenerateCommand.php]: final?

File: src/Command/SchemaCommand.php [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+use Cycle\ORM\Schema;
+use Cycle\ORM\SchemaInterface;
+use Spiral\Database\DatabaseManager;
+use Spiral\Migrations\Config\MigrationConfig;
+use Spiral\Migrations\Migrator;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Yiisoft\Yii\Console\ExitCode;
+
+class SchemaCommand extends BaseMigrationCommand
```
Comment (samdark) [inline src/Command/SchemaCommand.php]: final?

File: src/Command/UpCommand.php [hunk 7]
```diff
@@ -2,15 +2,34 @@
...
 use Spiral\Migrations\MigrationInterface;
+use Spiral\Migrations\Migrator;
+use Spiral\Migrations\State;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 use Yiisoft\Yii\Console\ExitCode;
+use Yiisoft\Yii\Cycle\Event\AfterMigrate;
+use Yiisoft\Yii\Cycle\Event\BeforeMigrate;
 
 class UpCommand extends BaseMigrationCommand
```
Comment (samdark) [inline src/Command/UpCommand.php]: final?

File: src/Logger/StdoutQueryLogger.php [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\LoggerTrait;
+use Psr\Log\LogLevel;
+
+/**
+ * Temporary LoggerInterface class
+ * Slightly adapted for SQL queries
+ * @package Yiisoft\Yii\Cycle\Logger
+ * @deprecated should be replaced
```
Comment (samdark) [inline src/Logger/StdoutQueryLogger.php]: ?

Comment (roxblnfk) [reply src/Logger/StdoutQueryLogger.php]: ¯\_(ツ)_/¯

**Comment by samdark** [comment]
Merged. Thanks!

**Comment by wolfy-j** [comment]
The Postgres PK issue has been addressed in 1.2.3

**Pull request merged**